rest-api/src/main/java/com/cloudbees/workflow/rest/external/RunExt.java

```diff
@@ -232,6 +232,7 @@ public static RunExt create(WorkflowRun run) {
 
                 if (StageNodeExt.isStageNode(node)) {
                     StageNodeExt stage = StageNodeExt.create(node);
+                    stage.addStageFlowNodes(node);
```

Comment:
This is needed currently since the StageExt serves itself from the appropriate RunExt cache, if present. 

THE CATCH is that this inflates the size of the RunExt object serialized and sent to clients.  

The UI could directly consume the fully-realized RunExt with full nested child nodes (and avoid calling requests to the FlowNodeAPI Describe API for each stage start node).

Until that happens, a way to avoid bloat on the response but still keep caching simple:
* Create a RunExtTrimmer class that wraps a RunExt, but removes the child nodes from StageExt objects returned (and saves them). 

Catch: you still have to do full stage processing up from on the RunExt.  Perhaps we should avoid that in cases where the Run is still building since it's expensive to do this.  But in other cases, you *will* have to do it at some point, might as well be up front.